deluge/tests/test_alertmanager.py

```diff
@@ -30,6 +55,31 @@ def handler(alert):
         self.am.register_handler('dummy_alert', handler)
         self.assertEqual(self.am.handlers['dummy_alert'], [handler])
 
+    def test_pop_alert(self):
+        def dummy_alert_1_handler(alert):
+            d.addCallback(self.assertEqual, alert.message, '1')
+            d.callback(None)
+
+        self.am.session = SessionMock()
+        self.am.register_handler('DummyAlert1', dummy_alert_1_handler)
+
+        d = Deferred()
+        return d
+
+    def test_pause_not_pop_alert(self):
+        def check_event(value):
+            event = self.am._event.isSet()
+            self.assertFalse(event)
+
+        def dummy_alert_1_handler(alert):
+            self.fail()
+
+        self.am.session = SessionMock()
```

Comment:
I think this should probably be put in set_up().

Reply:
I tried moving it to `set_up()`, but got errors on other tests because of it.
So currently I'm not changing it.

Reply:
4 years later, done 😅